core/loadpoint.go

```diff
@@ -595,8 +599,19 @@ func (lp *LoadPoint) publishSoC() {
 	if lp.SoC.AlwaysUpdate || lp.connected() {
 		f, err := lp.vehicle.ChargeState()
 		if err == nil {
-			lp.socCharge = f
-			lp.log.DEBUG.Printf("vehicle soc: %.0f%%", lp.socCharge)
+
+			if lp.SoC.EstimateSoC {
+				/* new soc value read from car api */
+				if f != lp.socChargeCar {
+					lp.socChargeCar = f
+					lp.chargedEnergySoc = lp.chargedEnergy
+				}
+				lp.socCharge = lp.socChargeCar + (lp.chargedEnergy - lp.chargedEnergySoc) / (float64(lp.vehicle.Capacity()) * 1e3) * 100
```

Comment:
Du hast eine Zeile zuvor lp.chargedEnergySoc = lp.chargedEnergy gesetzt. Daher ist lp.chargedEnergy - lp.chargedEnergySoc immer 0. Die Zeilen müssen getauscht werden, oder?

Müsste danach auch mit `lp.socCharge = lp.socChargeCar + ((lp.chargedEnergy - lp.chargedEnergySoc) * m)` klappen.

Reply:
Zeile 607 gilt nur beim Update vom Fahrzeug. In diesem Zeitpunkt muss die differnez Null sein, da ich den Fahrzeugwert weiterverwenden will

Reply:
Stimmt :)